scripts/build_bidding_zones.py

```diff
@@ -0,0 +1,195 @@
+# SPDX-FileCopyrightText: Contributors to PyPSA-Eur <https://github.com/pypsa/pypsa-eur>
+#
+# SPDX-License-Identifier: MIT
+import geopandas as gpd
+import pandas as pd
+
+
+def parse_zone_names(zone_names: pd.Series) -> tuple[set[str], pd.Series]:
+    """
+    Parse bidding zone names to extract country codes and handle combined zones.
+
+    Args:
+        zone_names: Series of zone names (e.g., 'DE', 'DE_LU', etc.)
+
+    Returns:
+        tuple containing:
+            - set of country codes covered by the zones
+            - Series of combined country strings (e.g., 'DE,LU')
+    """
+    # Extract primary country codes (first two characters)
+    primary_countries = set(zone_names.str[:2])
+
+    # Parse combined zones (e.g., 'DE_LU' -> 'LU')
+    secondary_countries = (
+        zone_names.str.split("_", expand=True)[1]
+        .dropna()
+        .loc[lambda x: x.str.len() == 2]
+    )
+
+    # Create combined country strings (e.g., 'DE,LU')
+    country_strings = zone_names.str[:2] + (
+        "," + secondary_countries.reindex(zone_names.index)
+    ).fillna("")
+
+    # Remove secondary countries from primary set
+    covered_countries = primary_countries | set(secondary_countries)
+
+    return covered_countries, country_strings
```

Comment:
This output could be removed as it is not used.

```suggestion
    return country_strings
```